dev/linters/shellcheck.sh

```diff
@@ -6,4 +6,5 @@ workspace=$(bazel info workspace)
 
 # shellcheck disable=SC2046
 # We want word splitting with find.
-bazel run @shellcheck//:binary -- $(find "${workspace}" -name '*.sh')
+bazel run @shellcheck//:binary -- \
+      $(find "${workspace}" -name '*.sh' | grep -v '/_work/')
```

Comment:
This command should ignore everything under `.gitignore` instead of hardcoding `_work` here.

Reply:
Now idea how to do that. Essentially a tool reading .gitignore, and a list of paths, and returning the filtered set of paths. ... Has git a subcommand for that ?

Reply:
Hm. https://git-scm.com/docs/gitignore mentions `git ls-files` and `git read-tree` ... Might be possible to use that, merging find+grep into one command.

Reply:
Generally accepted, will have to trial locally a bit.

Reply:
How about inverting the rationale here?
```
bazel run @shellcheck//:binary -- $(git ls-files | awk '/.sh$/ { print $0 }')
```

Reply:
Oh, when I commented, your comments were not visible yet. I'm glad that we though in the same way.

Reply:
Actually, you don't even need the `awk`:
```
bazel run @shellcheck//:binary -- $(git ls-files '*.sh')
```

Reply:
Shellcheck doesn't like relative paths. You will need:
```
bazel run @shellcheck//:binary -- \
  $(git ls-files --full-name '*.sh' | sed "s|^|${workspace}/|")
```

Reply:
Resolved with commit b087191de9161718ed1878a2aadc97d15f28d4e1